app/components/markdown/at_mention/at_mention.tsx

```diff
@@ -93,6 +101,47 @@ const AtMention = ({
         return user.mentionKeys;
     }, [currentUserId, mentionKeys, user]);
 
+    // Checks if the mention is a group
+    const group = useMemo(() => {
+        if (user?.username) {
+            return undefined;
+        }
+        const getGroupsByName = (gs: GroupModelType[]) => {
+            const groupsByName: Dictionary<GroupModelType> = {};
+
+            for (const g of gs) {
+                groupsByName[g.name] = g;
+            }
+
+            return groupsByName;
+        };
+
+        const groupsByName = getGroupsByName(groups);
+        let mn = mentionName.toLowerCase();
+
+        while (mn.length > 0) {
+            if (groupsByName[mn]) {
+                return groupsByName[mn];
+            }
+
+            // Repeatedly trim off trailing punctuation in case this is at the end of a sentence
+            if ((/[._-]$/).test(mn)) {
+                mn = mn.substring(0, mn.length - 1);
+            } else {
+                break;
+            }
+        }
+
+        // @ts-expect-error: The model constructor is hidden within WDB type definition
+        return new GroupModel(database.get(GROUP), {name: ''});
```

Comment:
Why do we need to create a group model here? Can't we just return undefined?

Reply:
Tbh I'm not entirely sure - I've followed exactly how `const user = useMemo(...` was written and I didn't want to deviate between the two, though an empty user / group model I can see is keeping things type safe further down, and we just use `if(user?.username` or `if(group?.name` to check if the model was found